ddtrace/internal/processor/trace.py

```diff
@@ -226,3 +227,28 @@ def shutdown(self, timeout):
         except ServiceStatusError:
             # It's possible the writer never got started in the first place :(
             pass
+
+
+@attr.s
+class SingleSpanSamplingProcessor(SpanProcessor):
+    """SpanProcessor for sampling single spans"""
+
+    rules = attr.ib(type=(List[SpanSamplingRule]))
+
+    def on_span_start(self, span):
+        # type: (Span) -> None
+        pass
+
+    def on_span_finish(self, span):
+        # type: (Span) -> None
+        # only sample if the span isn't already going to be sampled by trace sampler
+        if span.context.sampling_priority is not None and span.context.sampling_priority <= 0:
+            for rule in self.rules:
+                rule.sample(span)
+                # If we matched a rule, then don't try to apply any further rules
+                if rule.match(span):
+                    break
+
+    def shutdown(self, timeout):
```

Comment:
Should I be doing anything here?

Reply:
you don't have to override this.

it isn't marked as an abstract method so you'll just inherit the default (do nothing) behavior from the base class.